BLUESPAWN-win-client/src/util/processes/ParseCobalt.cpp

```diff
@@ -2,243 +2,237 @@
-					}
-				}
-				return wrapper;
-			} else{
-				return std::nullopt;
-			}
-		}
-		section = section.GetOffset(sizeof(IMAGE_SECTION_HEADER));
-	}
-	return std::nullopt;
-}
 
-std::optional<AllocationWrapper> FindBeaconInfo(const MemoryWrapper<>& memory){
-
-	// Experimentation shows that the configuration information is located 0x30 bytes into the .data section
-	// Check that before searching the entire module
-	auto quick{ FindBeaconInfoQuick(memory) };
-	if(quick){
-		return quick;
-	}
-
-	DWORD64 start{ 0x020001000100 };
-	DWORD64 xor3{ 0x696969696969 };
-	DWORD64 xor4{ 0x2E2E2E2E2E2E };
-	for(size_t idx = 0; idx < memory.MemorySize - 8; idx++){
-		auto pattern{ (*memory.GetOffset(idx).Convert<DWORD64>() & 0xFFFFFFFFFFFF) ^ start };
-		if(pattern == xor3 || pattern == xor4){
-			LOG_VERBOSE(2, "Beacon magic bytes found in memory");
-			auto wrapper{ memory.GetOffset(idx).ToAllocationWrapper(4096) };
-			for(size_t idx1 = 0; idx1 < wrapper.GetSize(); idx1++){
-				if(pattern == xor3){
-					wrapper.SetByte(idx1, wrapper[idx1] ^ 0x69);
-				} else{
-					wrapper.SetByte(idx1, wrapper[idx1] ^ 0x2E);
-				}
-			}
-			return wrapper;
-		}
-	}
-
-	return std::nullopt;
+std::optional<AllocationWrapper> FindBeaconInfoQuick(const MemoryWrapper<>& memory) {
+    IMAGE_DOS_HEADER hdr{};
+    auto offset{ memory.Convert<IMAGE_DOS_HEADER>()->e_lfanew };
+    if(memory.Convert<IMAGE_DOS_HEADER>()->e_magic != 0x5D4A) {
+        return std::nullopt;
+    }
+
+    auto section{ memory.GetOffset(offset + sizeof(IMAGE_NT_HEADERS)).Convert<IMAGE_SECTION_HEADER>() };
+    while(section->Name) {
+        if(*reinterpret_cast<PDWORD64>(section->Name) == *reinterpret_cast<PDWORD64>(".data\0\0")) {
+            DWORD dwRawDword{ *memory.GetOffset(section->PointerToRawData + 0x30).Convert<DWORD>() ^ 0x01000100 };
+            DWORD dwVirDword{ *memory.GetOffset(section->VirtualAddress + 0x30).Convert<DWORD>() ^ 0x01000100 };
+            if(dwRawDword == 0x2E2E2E2E || dwRawDword == 0x69696969) {
+                auto wrapper{
+                    memory.GetOffset(section->PointerToRawData + 0x30).Convert<DWORD>().ToAllocationWrapper(4096)
+                };
+                if(!wrapper) {}
+                for(size_t idx1 = 0; idx1 < wrapper.GetSize(); idx1++) {
+                    if(dwRawDword == 0x69696969) {
+                        wrapper.SetByte(idx1, wrapper[idx1] ^ 0x69);
+                    } else {
+                        wrapper.SetByte(idx1, wrapper[idx1] ^ 0x2E);
+                    }
+                }
+                return wrapper;
+            } else if(dwRawDword == 0x2E2E2E2E || dwRawDword == 0x69696969) {
+                auto wrapper{
+                    memory.GetOffset(section->PointerToRawData + 0x30).Convert<DWORD>().ToAllocationWrapper(4096)
+                };
+                for(size_t idx1 = 0; idx1 < wrapper.GetSize(); idx1++) {
+                    if(dwRawDword == 0x69696969) {
```

Comment:
Codacy found an issue: [Condition 'dwRawDword==1768515945' is always false](https://app.codacy.com/manual/ION28/BLUESPAWN/pullRequest?prid=6145581)